src/ray/object_manager/pull_manager.cc

```diff
@@ -583,6 +583,28 @@ bool PullManager::PullRequestActiveOrWaitingForMetadata(uint64_t request_id) con
   return bundle_it->second.num_object_sizes_missing > 0;
 }
 
+bool PullManager::QueueAtCapacity(const Queue &bundles,
+                                  uint64_t highest_req_id_being_pulled) const {
+  auto last_request_it = bundles.find(highest_req_id_being_pulled);
+  if (last_request_it == bundles.end()) {
+    // No requests are active.
+    return false;
+  }
+
+  last_request_it++;
+  if (last_request_it == bundles.end()) {
+    // No other requests in the queue.
+    return false;
+  }
+
+  return true;
+}
+
+bool PullManager::AtCapacity() const {
```

Comment:
Is the following an equivalent simpler condition?

`object_pull_requests_.size() != active_object_pull_requests_.size()`

Reply:
Actually I'm realizing that neither of these is sufficient because we may also block pulling when we don't have an object size yet.